src/inventory.h

```diff
@@ -40,13 +40,15 @@ struct ItemStack
 	~ItemStack() = default;
 
 	// Serialization
-	void serialize(std::ostream &os) const;
+	void serialize(std::ostream &os, bool serialize_meta = true) const;
 	// Deserialization. Pass itemdef unless you don't want aliases resolved.
 	void deSerialize(std::istream &is, IItemDefManager *itemdef = NULL);
 	void deSerialize(const std::string &s, IItemDefManager *itemdef = NULL);
 
 	// Returns the string used for inventory
 	std::string getItemString() const;
```

Comment:
I suggest `std::string getItemString(bool include_metadata=false) const;`, and removing `dump()`

Reply:
The default needs to be true to not break existing code

Reply:
correct, I made a mistake with the default there

Reply:
IMO Human readable formatting and machine formatting are best kept separate (as functions) to avoid calling the wrong one by accident.
I'm not requesting any changes; just wondered about this request.